Gemfile

```diff
@@ -6,40 +6,49 @@ version = ENV["RAILS_VERSION"] || "4.2"
 
 if version == 'master'
   gem 'rack', github: 'rack/rack'
+  gem 'arel', github: 'rails/arel'
+  gem 'rails', github: 'rails/rails'
   git 'https://github.com/rails/rails.git' do
     gem 'railties'
     gem 'activesupport'
     gem 'activemodel'
     gem 'actionpack'
+    gem 'activerecord', group: :test
     # Rails 5
     gem 'actionview'
   end
   # Rails 5
   gem 'rails-controller-testing', github: 'rails/rails-controller-testing'
 else
   gem_version = "~> #{version}.0"
+  gem 'rails', gem_version
   gem 'railties', gem_version
   gem 'activesupport', gem_version
   gem 'activemodel', gem_version
   gem 'actionpack', gem_version
+  gem 'activerecord', gem_version, group: :test
 end
 
 if RUBY_VERSION < '2'
   gem 'mime-types', [ '>= 2.6.2', '< 3' ]
 end
 
+if RUBY_VERSION < '2.1'
+  gem 'nokogiri', '< 1.7'
```

Comment:
This was the original change, but other things were still broken, so I moved over the relevant setup from master